pinot-core/src/test/java/com/linkedin/pinot/core/common/RealtimeNoDictionaryTest.java

```diff
@@ -0,0 +1,269 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.linkedin.pinot.core.common;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Random;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+import com.linkedin.pinot.common.data.FieldSpec;
+import com.linkedin.pinot.common.data.MetricFieldSpec;
+import com.linkedin.pinot.core.io.readerwriter.impl.FixedByteSingleColumnSingleValueReaderWriter;
+import com.linkedin.pinot.core.operator.BaseOperator;
+import com.linkedin.pinot.core.realtime.impl.datasource.RealtimeColumnDataSource;
+
+
+public class RealtimeNoDictionaryTest {
```

Comment:
+1 for adding unit tests. Are there tests that cover the query execution path as well?